negative_test/test_hanging_xlog.py

```diff
@@ -0,0 +1,39 @@
+from pathlib import Path
+
+from lib import Options
+from lib.log_helper import Log
+
+
+class TestHangingXLog:
+
+    def test_hanging_xlog(self, test_run_log: Log) -> None:
+        """Check if tarantool is hanging it will be stopped by start_timeout.
+
+        Check next patterns in the test-run log:
+        There are (timeout // 10 - 1) lines with 'No output' warnings, which
+        are going sequentially.
+        There are no (timeout // 10)th line with same warning.
+        Next, find log line with failing on start server timeout.
+        Finally, check log contains '[ fail ]'.
+        """
```

Comment:
My expectation was that you'll describe what is going on in the test. Let's imagine I'm a stranger: what I see here? hang.test.lua starts replica.lua (which should use hang.test.lua as master). I asking myself: 'what is wrong here?', but there is no answer. Why the test should hang?

IOW, I want the story being persisted somewhere. Like so:

* hang.test.lua starts replica.lua.
* replica.lua calls box.cfg{replication = {<..hang.test.lua listen port..>}}.
* replica.lua unable to bootstrap, so test-run will never see 'entering the event loop' log entry.
* Before the fix: test-run spins in a kind of 'busy loop' around grepping tarantool's log and test timeout have no chance to occur.
* After the fix: test-run spins around grepping the log, but gently pass execution flow for other greenlets. So, test timeout will work as expected.
* After introducing start server timeout: we'll hit this new timeout before the test timeout, but everything else is the same.

Reply:
In fact, any other way to describe what is going on is okay too. We should not miss this knowledge, that is my point.

Reply:
Not relevant anymore.